src/main/java/com/google/sps/servlets/DataFromDatabase.java

```diff
@@ -0,0 +1,134 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/data-from-db")
+public class DataFromDatabase extends HttpServlet {
+
+  DatabaseClient dbClient;
+  private String[] selectedTables;
+
+  public void init() {
+    Spanner spanner = SpannerOptions.newBuilder().build().getService();
+    DatabaseId db = DatabaseId.of("play-user-data-beetle", "test-instance", "example-db");
+    this.dbClient = spanner.getDatabaseClient(db);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+
+    for (String table : selectedTables) {
+      String colQuery = "SELECT column_name, spanner_type, is_nullable FROM information_schema.columns WHERE table_name = '" + table + "'";
+
+      try (ResultSet resultSet =
+        dbClient.singleUse().executeQuery(Statement.of(colQuery))) {
+
+        List<String> colnames = new ArrayList<>();
+        List<String> spannerTypes = new ArrayList<>();
+        List<String> nullables = new ArrayList<>();
+
+        while (resultSet.next()) {
+          colnames.add(resultSet.getString(0));
+          spannerTypes.add(resultSet.getString(1));
+          nullables.add(resultSet.getString(2));
+        }
+
+        String query = "SELECT ";
+        for (String colName : colnames) {
+          query += (colName + ", ");
+        }
+        query = query.substring(0, query.length() - 2);
+        query += " FROM " + table; 
+
+        response.getWriter().println("Table: " + table);
+
+        executeTableQuery(query, colnames, spannerTypes, response);
+      }
+    }
+  }
+
+  private void executeTableQuery(String query, List<String> colnames, List<String> spannerTypes, HttpServletResponse response) throws IOException {
+    try (ResultSet rs =
+      dbClient
+      .singleUse() // Execute a single read or query against Cloud Spanner.
+      .executeQuery(Statement.of(query))) {
+
+      while (rs.next()) {
+        String row = "";
+        int index = 0;
+        while (index < colnames.size()) {
```

Comment:
nit: this seems like an appropriate time to use a for loop

Reply:
Done